features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -461,17 +461,30 @@ ssize_t ATHandler::read(char *buf, size_t size, bool read_even_stop_tag, bool he
     }
 
     size_t match_pos = 0;
-    size_t read_size = hex ? size * 2 : size;
+    // Size includes NULL
+    size_t read_size = hex ? (size - 1) * 2 : (size - 1);
 
     consume_char('\"');
 
+    if (_last_err) {
```

Comment:
Why is this check here if it's also being checked 10 lines above?

Reply:
consume_char() could trigger error if there is nothing to read. In that case if size is 1, this routine would return 0, but it is true that _last_err would be set to error in that case(by consume_char()). I could skip this check and change the unit test to be fine with return value 0 and _last_err set to error.